helix-term/src/commands.rs

```diff
@@ -2934,6 +2934,27 @@ pub mod cmd {
         Ok(())
     }
 
+    fn run_shell_command(
+        cx: &mut compositor::Context,
+        args: &[Cow<str>],
+        _event: PromptEvent,
+    ) -> anyhow::Result<()> {
+        let shell = &cx.editor.config.shell;
+        let (output, _success) = match shell_impl(shell, &args.join(" "), None) {
+            Ok(result) => result,
+            Err(err) => {
+                cx.editor.set_error(err.to_string());
+                return Ok(());
+            }
+        };
```

Comment:
```suggestion
        let (output, _success) = match shell_impl(shell, &args.join(" "), None)?;
```

TypedCommand takes care of error handling and printing the error to the statusline.

Reply:
Sure, I deleted `match`